### ISSUE TITLE
jekyll-seo prevents generation of site

### ISSUE THREAD

**Issue by aliou** [double-great/great-great-jekyll-theme#35]
Hello! Following #34, I tried installing great-great-jekyll-theme in a new Jekyll blog. However, I run into this issue:
```
Liquid Exception: Liquid syntax error (/private/var/folders/2w/pnktcdh12h90f262tg8b9f1c0000gn/T/jekyll-remote-theme-20201218-32417-1kt0xth/_includes/head.html line 4):
Unknown tag 'seo' included in /private/var/folders/2w/pnktcdh12h90f262tg8b9f1c0000gn/T/jekyll-
remote-theme-20201218-32417-1kt0xth/_layouts/default.html
bundler: failed to load command: jekyll (/usr/local/var/rbenv/versions/2.6.6/bin/jekyll)
...
/usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/document.rb:23:in `unknown_tag':
Liquid syntax error (/private/var/folders/2w/pnktcdh12h90f262tg8b9f1c0000gn/T/jekyll-remote-theme-20201218-32417-1kt0xth/_includes/head.html line 4):
Unknown tag 'seo' included  (Liquid::SyntaxError)
```

<details>
<summary>Full backtrace:</summary>
<pre>
<code>$ bundle exec jekyll build --trace
Configuration file: /Users/alioudiallo/code/tmp/test/_config.yml
      Remote Theme: Using theme double-great/great-great-jekyll-theme
            Source: /Users/alioudiallo/code/tmp/test
       Destination: /Users/alioudiallo/code/tmp/test/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
      Remote Theme: Using theme double-great/great-great-jekyll-theme
       Jekyll Feed: Generating feed for posts
  Liquid Exception: Liquid syntax error (/private/var/folders/2w/pnktcdh12h90f262tg8b9f1c0000gn/T/jekyll-remote-theme-20201218-32980-ic04yp/_includes/head.html line 4): Unknown tag 'seo' included in /private/var/folders/2w/pnktcdh12h90f262tg8b9f1c0000gn/T/jekyll-remote-theme-20201218-32980-ic04yp/_layouts/default.html
bundler: failed to load command: jekyll (/usr/local/var/rbenv/versions/2.6.6/bin/jekyll)
Traceback (most recent call last):
        69: from /usr/local/var/rbenv/versions/2.6.6/bin/bundle:23:in `<main>'
        68: from /usr/local/var/rbenv/versions/2.6.6/bin/bundle:23:in `load'
        67: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/exe/bundle:37:in `<top (required)>'
        66: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/friendly_errors.rb:130:in `with_friendly_errors'
        65: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/exe/bundle:49:in `block in <top (required)>'
        64: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/cli.rb:24:in `start'
        63: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        62: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/cli.rb:30:in `dispatch'
        61: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        60: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        59: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        58: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/cli.rb:497:in `exec'
        57: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/cli/exec.rb:28:in `run'
        56: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/cli/exec.rb:63:in `kernel_load'
        55: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/bundler-2.2.0/lib/bundler/cli/exec.rb:63:in `load'
        54: from /usr/local/var/rbenv/versions/2.6.6/bin/jekyll:23:in `<top (required)>'
        53: from /usr/local/var/rbenv/versions/2.6.6/bin/jekyll:23:in `load'
        52: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/exe/jekyll:15:in `<top (required)>'
        51: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
        50: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
        49: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
        48: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
        47: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
        46: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
        45: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
        44: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `each'
        43: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
        42: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:36:in `process'
        41: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:65:in `build'
        40: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:28:in `process_site'
        39: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:80:in `process'
        38: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:210:in `render'
        37: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:530:in `render_docs'
        36: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:530:in `each_value'
        35: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:531:in `block in render_docs'
        34: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:531:in `each'
        33: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:532:in `block (2 levels) in render_docs'
        32: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:547:in `render_regenerated'
        31: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:63:in `run'
        30: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:93:in `render_document'
        29: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:163:in `place_in_layouts'
        28: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:194:in `render_layout'
        27: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:131:in `render_liquid'
        26: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:36:in `render!'
        25: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:70:in `measure_time'
        24: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:37:in `block in render!'
        23: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:63:in `measure_bytes'
        22: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:38:in `block (2 levels) in render!'
        21: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:59:in `measure_counts'
        20: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:39:in `block (3 levels) in render!'
        19: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:220:in `render!'
        18: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:207:in `render'
        17: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:242:in `with_profiling'
        16: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:208:in `block in render'
        15: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:91:in `render'
        14: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:103:in `render_node_to_output'
        13: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/tags/include.rb:206:in `render'
        12: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/context.rb:123:in `stack'
        11: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/tags/include.rb:208:in `block in render'
        10: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/inclusion.rb:15:in `render'
         9: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:12:in `parse'
         8: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:70:in `measure_time'
         7: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:13:in `block in parse'
         6: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:116:in `parse'
         5: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:132:in `parse'
         4: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/document.rb:5:in `parse'
         3: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/document.rb:10:in `parse'
         2: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:32:in `parse'
         1: from /usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/document.rb:11:in `block in parse'
/usr/local/var/rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/document.rb:23:in `unknown_tag': Liquid syntax error (/private/var/folders/2w/pnktcdh12h90f262tg8b9f1c0000gn/T/jekyll-remote-theme-20201218-32980-ic04yp/_includes/head.html line 4): Unknown tag 'seo' included  (Liquid::SyntaxError)</code>
</pre>
</details>

It seems like Liquid doesn't know the [`seo`](https://github.com/double-great/great-great-jekyll-theme/blob/e705057a8ee6d795278b05cab61c56f96444d4f2/_includes/head.html#L4) tag when generating a layout including the `head` include.

I tried directly adding the `jekyll-seo` plugin in the `jekyll_plugins` block in the Gemfile and in the array of plugins in `_config.yml` but I have the same issue.

I took a look at the [double-great.github.io](https://github.com/double-great/double-great.github.io) which uses the theme but I can build the blog locally without any issue.

Do you know if I'm missing something to make this work with a new blog? 

Thanks!

**Comment by jsnmrs** [comment]
Hi @aliou!

I created a reduced test case for a new blog, with the minimal information required for a successful build. This is using the `github-pages` gem (Jekyll v3.9.0), instead of `jekyll` (Jekyll v4.2.0).

**config.yml**
```
title: Test
description: "Test site."
github: example/test

remote_theme: double-great/great-great-jekyll-theme@main

plugins:
  - jekyll-feed
  - jekyll-sitemap
  - jekyll-seo
  - jekyll-remote-theme

```

**Gemfile**
```
source "https://rubygems.org"
gem "github-pages"  , group: :jekyll_plugins
```

**index.md**
```
---
layout: default
---
```

I'm using Ruby `2.7.0p0` and bundler `2.2.2`. After running `bundle install`, `bundle exec github-pages versions` shows me:
```
+------------------------------+---------+
| Gem                          | Version |
+------------------------------+---------+
| jekyll                       | 3.9.0   |
| jekyll-sass-converter        | 1.5.2   |
| kramdown                     | 2.3.0   |
| kramdown-parser-gfm          | 1.1.0   |
| jekyll-commonmark-ghpages    | 0.1.6   |
| liquid                       | 4.0.3   |
| rouge                        | 3.23.0  |
| github-pages-health-check    | 1.16.1  |
| jekyll-redirect-from         | 0.16.0  |
| jekyll-sitemap               | 1.4.0   |
| jekyll-feed                  | 0.15.1  |
| jekyll-gist                  | 1.5.0   |
| jekyll-paginate              | 1.1.0   |
| jekyll-coffeescript          | 1.1.1   |
| jekyll-seo-tag               | 2.6.1   |
| jekyll-github-metadata       | 2.13.0  |
| jekyll-avatar                | 0.7.0   |
| jekyll-remote-theme          | 0.4.2   |
| jemoji                       | 0.12.0  |
| jekyll-mentions              | 1.6.0   |
| jekyll-relative-links        | 0.6.1   |
| jekyll-optional-front-matter | 0.3.2   |
| jekyll-readme-index          | 0.3.0   |
| jekyll-default-layout        | 0.1.4   |
| jekyll-titles-from-headings  | 0.5.3   |
| jekyll-swiss                 | 1.0.0   |
| minima                       | 2.5.1   |
| jekyll-theme-primer          | 0.5.4   |
| jekyll-theme-architect       | 0.1.1   |
| jekyll-theme-cayman          | 0.1.1   |
| jekyll-theme-dinky           | 0.1.1   |
| jekyll-theme-hacker          | 0.1.2   |
| jekyll-theme-leap-day        | 0.1.1   |
| jekyll-theme-merlot          | 0.1.1   |
| jekyll-theme-midnight        | 0.1.1   |
| jekyll-theme-minimal         | 0.1.1   |
| jekyll-theme-modernist       | 0.1.1   |
| jekyll-theme-slate           | 0.1.1   |
| jekyll-theme-tactile         | 0.1.1   |
| jekyll-theme-time-machine    | 0.1.1   |
+------------------------------+---------+
```

Running `bundle exec jekyll build --trace` runs without error.

If you're continuing to see errors, can you post your `config.yml` and `Gemfile`?

**Comment by aliou** [comment]
Hello @jsnmrs!

I've copied the `_config.yml` and `Gemfile` files below. The only difference I'm seeing is the `github-pages` gem; indeed replacing the content of the Gemfile only by the `github-pages` gem fixes the generation.

#### `_config.yml`
```yml
title: Your awesome title
email: your-email@example.com
description: >- # this means to ignore newlines until "baseurl:"
  Write an awesome description for your new site here. You can edit this
  line in _config.yml. It will appear in your document head meta (for
  Google search results) and in your feed.xml site description.
baseurl: "" # the subpath of your site, e.g. /blog
url: "" # the base hostname & protocol for your site, e.g. http://example.com
twitter_username: jekyllrb
github_username:  jekyll

# Build settings
# theme: minima
plugins:
  - jekyll-feed
  - jekyll-remote-theme

remote_theme: double-great/great-great-jekyll-theme@main
```

#### `Gemfile`
```ruby
source "https://rubygems.org"

gem "jekyll", "~> 4.2.0"

group :jekyll_plugins do
  gem "jekyll-feed", "~> 0.12"
  gem "jekyll-remote-theme"
end
```

**Comment by jsnmrs** [comment]
Hi @aliou — the fix for this issue is in #37.

If using the SEO plugin, change the entry in `config.yml` from `jekyll-seo` to `jekyll-seo-tag`

With that change, a functional `Gemfile` that does not use the `github-pages` gem is:
```
gem "jekyll"
gem "kramdown-parser-gfm"
group :jekyll_plugins do
  gem "jekyll-feed"
  gem "jekyll-remote-theme"
  gem "jekyll-seo-tag"
  gem "jekyll-sitemap"
end
```
